olive/engine/engine.py

```diff
@@ -138,16 +138,16 @@ def initialize(self):
         """
         Initialize engine state. This should be done before running the registered passes.
         """
-        cache_dir = self._config.cache_dir
+        self.cache_dir = self._config.cache_dir
```

Comment:
do we need make cache_dir as part of the engine attribute? it seems the save_model still use self._config.cache_dir?

Reply:
Oh yeah, I forgot about `self._config.cache_dir`. I will revert this change and use `self._config.cache_dir` in the `cache_utils.get_non_local_resource` call.

Reply:
Done